File: for-doc-java/src/main/java/org/doc/j/model/Book.java
package org.doc.j.model;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.math.BigDecimal;
import java.util.List;

@Entity
public interface Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    String name();

    int edition();

    BigDecimal price();

    @Nullable
    @ManyToOne
    BookStore store();

    @ManyToMany
    List<Author> authors();
}

File: for-doc-kotlin/src/main/kotlin/org/doc/k/model/Book.kt
package org.doc.k.model

import org.babyfish.jimmer.sql.*

@Entity
interface Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    val name: String

    val edition: Int

    @ManyToOne
    val store: BookStore?

    @ManyToMany
    val authors: List<Author>
}